site/content/blog/simplifying-package-authoring.md

```diff
@@ -0,0 +1,301 @@
+
+5:02:03PM: ---- applying 2 changes [0/3 done] ----
+5:02:03PM: create clusterrole/dynatrace-sa-role (rbac.authorization.k8s.io/v1) cluster
+5:02:03PM: create serviceaccount/dynatrace-sa (v1) namespace: default
+5:02:03PM: ---- waiting on 2 changes [0/3 done] ----
+...
+5:02:03PM: ok: reconcile clusterrolebinding/dynatrace-sa-role-binding (rbac.authorization.k8s.io/v1) cluster
+5:02:03PM: ---- applying complete [3/3 done] ----
+5:02:03PM: ---- waiting complete [3/3 done] ----
+
+Succeeded
+$ kubectl create ns dynatrace
+namespace/dynatrace created
+```
+
+Let's deploy with `kctrl dev`:
+
+As all the manifest are sync'ed locally by `pkg init` command, we have to provide the `--local` flag also.
+
+```bash
+$ kctrl dev -f package-resources.yml --local
+Target cluster 'https://192.168.64.105:8443' (nodes: minikube)
+
+apiVersion: packaging.carvel.dev/v1alpha1
+kind: PackageInstall
+metadata:
+  annotations:
+    kctrl.carvel.dev/local-fetch-0: .
+  creationTimestamp: null
+  name: dynatrace
+  namespace: default
+spec:
+  packageRef:
+    refName: dynatrace.carvel.dev
+    versionSelection:
+      constraints: 0.0.0
+  serviceAccountName: dynatrace-sa
+status:
+  conditions: null
+  friendlyDescription: ""
+  observedGeneration: 0
+
+Reconciling in-memory app/dynatrace (namespace: default) ...
+==> Executing /usr/local/bin/vendir [vendir sync -f - --lock-file /dev/null]
+==> Finished executing /usr/local/bin/vendir
+
+==> Executing /usr/local/bin/ytt [ytt -f /var/folders/z1/czwbshk524j7t12s3tqhc_100000gn/T/kapp-controller-fetch-template-deploy3298499314/0/upstream]
+==> Finished executing /usr/local/bin/ytt
+
+==> Executing /usr/local/bin/kbld [kbld -f - --build=false]
+==> Finished executing /usr/local/bin/kbld
+
+==> Executing /usr/local/bin/kapp [kapp deploy --prev-app dynatrace-ctrl -f - --app-changes-max-to-keep=5 --app dynatrace.app --kubeconfig=/dev/null --yes]
+
+...
+	    | 5:11:27PM: ---- applying complete [23/23 done] ----
+	    | 5:11:27PM: ---- waiting complete [23/23 done] ----
+	    | Succeeded
+5:11:27PM: Deploy succeeded 
+
+Succeeded
+
+$ kubectl get pods -n dynatrace
+NAME                                  READY   STATUS    RESTARTS   AGE
+dynatrace-operator-775fcdb79f-f79pj   1/1     Running   0          4m43s
+dynatrace-webhook-6df6fc6f6c-4qbg7    1/1     Running   0          4m43s
+dynatrace-webhook-6df6fc6f6c-7dqkz    1/1     Running   0          4m43s
+```
+
+As we can see that `dynatrace-operator` pod is up, it means the package is behaving as expected. Now, we will use `kctrl pkg release` to create the package which will be released.
+
+### kctrl pkg release
+
+```bash
+$ kctrl pkg release -v 1.0.0
+```
+
+This command will create an imgpkg bundle, upload it to the OCI registry and create `package.yml` and `metadata.yml` which can be released for consumption eventually. In this command, it will ask only 1 question about where to push our imgpkg bundle. Also, we are versioning our package with `-v` flag.
```

Comment:
```suggestion
This command will create an imgpkg bundle, upload it to the OCI registry and create `package.yml` and `metadata.yml` which can be released for consumption eventually. In this command, it will ask only one question about where to push our imgpkg bundle. Also, we are versioning our package with `-v` flag.
```

Reply:
Adopted